quickcheck_macros/Cargo.toml

```diff
@@ -14,8 +14,12 @@ autotests = false
 [lib]
 name = "quickcheck_macros"
 path = "src/lib.rs"
-plugin = true
+proc-macro = true
 
-[dependencies.quickcheck]
-path = ".."
-version = "0.7.1"
+[dependencies]
+proc-macro2 = "0.4.19"
+quote = "0.6.8"
+syn = { version = "0.15", features = ["full"] }
+
+[dev-dependencies]
+quickcheck = { path = "..", version = "0.7.2" }
```

Comment:
Is it standard for macro crates to not depend directly on the crate they invoke?

Reply:
There are no API guidelines about this but that's how, for example, [serde_derive](https://github.com/serde-rs/serde/blob/a8a54c05681168d09dbdbb4212c31a142c3cdae4/serde_derive/Cargo.toml#L32), [failure_derive](https://github.com/rust-lang-nursery/failure/blob/8f8f92fa55e99d95a886cf54ad877f47059c7375/failure_derive/Cargo.toml#L18-L20) and [rocket_codegen](https://github.com/SergioBenitez/Rocket/blob/master/core/codegen/Cargo.toml#L28) behave.

The reasoning behind this is because macros only generate code, so they don't need the crate(s) they invoke to work. And depending on these crates won't gain any benefit --- users of the macros need to directly depend on those crates anyway.